hydromt/test_utils/test_equal.py

```diff
@@ -0,0 +1,69 @@
+"""Test utilities for comparing model components."""
+
+from typing import Tuple
+
+import xarray as xr
+from geopandas.testing import assert_geodataframe_equal
+
+from hydromt.components.base import ModelComponent
+from hydromt.components.grid import GridComponent
+from hydromt.components.region import ModelRegionComponent
+from hydromt.components.vector import VectorComponent
+from hydromt.models.model import Model
+
+
+def check_models_equal(one: Model, other: Model) -> Tuple[bool, dict[str, str]]:
+    """Test if two models including their data components are equal.
+
+    Parameters
+    ----------
+    other : Model (or subclass)
+        Model to compare against
+
+    Returns
+    -------
+    equal: bool
+        True if equal
+    errors: dict
+        Dictionary with errors per model component which is not equal
+    """
+    assert isinstance(other, type(one))
+    components = list(one._components.keys())
+    components_other = list(other._components.keys())
+    assert components == components_other
+    errors: dict[str, str] = {}
+    is_equal = True
+    for c in components:
+        component_equal, component_errors = check_components_equal(
+            getattr(one, c), getattr(other, c), c
+        )
+        is_equal &= component_equal
+        errors.update(**component_errors)
+    return is_equal, errors
+
+
+def check_components_equal(
```

Comment:
We will have to think this through later. How can we ensure that other components created by plugins are also testable.

Reply:
I suggest adding a test_equal method do each component and then keep the test_equal method in Model to loop over components.

Reply:
Discussed with the team. We will make the `test_equal` functionality a separate module for testing utils.